serverless/src/api/municipalities/createMunicipalitiesStats/index.js

```diff
@@ -0,0 +1,189 @@
+const {
+  getAllMunicipalitiesWithUsers,
+  getAllMunicipalities,
+} = require('../../../shared/municipalities');
+const { getMunicipalityGoal } = require('../../../shared/utils');
+
+const timePassed = 6 * 60 * 60 * 1000;
+const amountOfWins = 15;
+const amountOfNewcomers = 15;
+const amountOfChanged = 10;
+const changeThresholds = { absolute: 2000, relative: 30, population: 40000 };
+
+module.exports.handler = async event => {
+  try {
+    // No query param was passed, therefore we get all municipalities for which people
+    // have already signed up and compute the stats
+    const userMuncipality = await getAllMunicipalitiesWithUsers();
+
+    const statsWithAllMunicipalities = await computeStats({
+      userMuncipality,
+      shouldSendAllMunicipalities: true,
+    });
+
+    const statsWithEvents = await computeStats({
+      userMuncipality,
+      shouldSendAllMunicipalities: false,
+    });
+
+    await saveJson(statsWithAllMunicipalities, 'statsWithAll.json');
+    await saveJson(statsWithEvents, 'statsWithEvents.json');
+
+    return event;
+  } catch (error) {
+    console.log('error creating stats for places', error);
+    return errorResponse(500, 'Error while getting stats for places', error);
+  }
+};
+
+const computeStats = async ({
+  userMuncipality,
+  shouldSendAllMunicipalities,
+}) => {
+  const date = new Date();
+  const wins = [];
+  let newcomers = [];
+  let relativeChangers = [];
+  let absoluteChangers = [];
+  const municipalitiesWithUsers = [];
+  console.time('municipalities');
+  const municipalityMap = new Map();
+  for (const { userId, ags, createdAt, population } of userMuncipality) {
+    if (!municipalityMap.has(ags)) {
+      municipalityMap.set(ags, {
+        users: [{ userId, createdAt }],
+        population,
+      });
+    } else {
+      municipalityMap.get(ags).users.push({ userId, createdAt });
+    }
+  }
+  console.timeEnd('municipalities');
+
+  // If we should return all municipalities, we get them
+  // from the municipalities table, loop through and add the signups by mapping the array
+  if (shouldSendAllMunicipalities) {
+    const allMunicipalities = await getAllMunicipalities();
+
+    const allMunicipalitiesWithStats = allMunicipalities.map(municipality => {
+      const goal = getMunicipalityGoal(municipality.population);
+      const { ags } = municipality;
+
+      let signups = 0;
+      if (municipalityMap.has(municipality.ags)) {
+        signups = municipalityMap.get(municipality.ags).users.length;
+      }
+
+      return { ags, goal, signups };
+    });
+    return { municipalities: allMunicipalitiesWithStats };
+  }
+
+  for (const [ags, { users, population }] of municipalityMap) {
+    const goal = getMunicipalityGoal(population);
+
+    const filteredUsers = users.filter(
+      user => date - new Date(user.createdAt) < timePassed
+    );
+
+    const previous = users.length - filteredUsers.length;
+    const current = users.length;
+
+    if (current > goal) {
```

Comment:
if (current >= goal && previous < goal)

--> sonst sind immer die gleichen Gemeinden in der Kategorie